File: TenMinsJava/day87.java
// ask for an email & name input, if email is duplicate, update name(Update)
package TenMinsJava;

import java.io.File;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class day87 {

    public static void email(String email, String name) {
        File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
        ArrayList<String> lines = new ArrayList<>();
        boolean emailFound = false;

        try (Scanner scan = new Scanner(file)) {

            while (scan.hasNextLine()) {
                String line = scan.nextLine();
                String[] parts = line.split(",");

                if (parts[0].trim().equals(email)) {
                    // update name if email exists
                    lines.add(email + "," + name);
                    emailFound = true;
                } else {
                    lines.add(line);
                }
            }

            if (!emailFound) {
                if (!email.matches(".+@.+\\..+")) {
                    System.out.println("Invalid email format");
                    return;
                }
                lines.add(email + "," + name);
                System.out.println("Email saved successfully");
            } else {
                System.out.println("Duplicate email found. Name updated.");
            }

            // rewrite file
            FileWriter writer = new FileWriter(file);
            for (String l : lines) {
                writer.write(l + "\n");
            }
            writer.close();

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
}

File: TenMinsPy/day87.py
# ask for an email & name input, if email is duplicate, update name(Update)

import re

def checkEmail(email, name):
    pattern = r".+@.+\..+"
    file_path = "D:/coding30minsAday/textFiles/JTrialCsv.csv"

    if not re.match(pattern, email):
        print("nah man")
        return

    lines = []
    found = False

    with open(file_path, "r") as f:
        for line in f:
            if line.startswith(email + ","):
                lines.append(f"{email}, {name}\n")  # update
                found = True
            else:
                lines.append(line)

    with open(file_path, "w") as f:
        f.writelines(lines)

    if not found:
        with open(file_path, "a") as f:
            f.write(f"{email}, {name}\n")
        print("Email saved successfully")
    else:
        print("Duplicate email found — name updated")